device/alpaka/src/utils/get_vecmem_resource.cpp

```diff
@@ -0,0 +1,137 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2025 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Local include(s).
+#include "utils.hpp"
+
+// Project include(s).
+#include "traccc/alpaka/utils/get_vecmem_resource.hpp"
+
+// Standard library include(s)
+#include <memory>
+#include <stdexcept>
+
+namespace traccc::alpaka::details {
+
+// Per-accelerator types for vecmem.
+template <typename T>
+struct host_device_types {
+    using device_memory_resource = vecmem::host_memory_resource;
+    using host_memory_resource = vecmem::host_memory_resource;
+    using managed_memory_resource = vecmem::host_memory_resource;
+    using device_copy = vecmem::copy;
+    using device_async_copy = vecmem::copy;
+};
+template <>
+struct host_device_types<::alpaka::TagGpuCudaRt> {
+    using device_memory_resource = vecmem::cuda::device_memory_resource;
+    using host_memory_resource = vecmem::cuda::host_memory_resource;
+    using managed_memory_resource = vecmem::cuda::managed_memory_resource;
+    using device_copy = vecmem::cuda::copy;
+    using device_async_copy = vecmem::cuda::async_copy;
+};
+template <>
+struct host_device_types<::alpaka::TagGpuHipRt> {
+    using device_memory_resource = vecmem::hip::device_memory_resource;
+    using host_memory_resource = vecmem::hip::host_memory_resource;
+    using managed_memory_resource = vecmem::hip::managed_memory_resource;
+    using device_copy = vecmem::hip::copy;
+    using device_async_copy = vecmem::hip::copy;
```

Comment:
When we upgrade vecmem to pull in the new version, I can update this to actually use `async` for HIP too.

My follow up PR will be Async-Alpaka, so zero rush / can be made then.